api/index.js

```diff
@@ -12,24 +12,73 @@ const ews = require('ews-javascript-api');
 const azureStorage = require('azure-storage');
 const sha256 = require('sha256');
 const intoStream = require('into-stream');
+const jsonwebtoken = require('jsonwebtoken');
+const jsrsasign = require('jsrsasign');
+const url = require('url');
+const uuidv4 = require('uuid/v4');
 
 var utils = require('../utils');
-
 var app = express();
 
 const iberaServicesEndpoint = config.IBERA_SERVICES_ENDPOINT;
 const azureStorageConnectionString = config.STORAGE_CONNECTION_STRING;
 
-const USER_TOKEN_HEADER_KEY = 'user-token';
+const USER_ACCOUNT_HEADER_KEY = 'user-account';
 const CONTAINER_NAME = 'attachments';
 
 var azureBlobService = azureStorage.createBlobService(azureStorageConnectionString);
 
-//Will be implemented next,  https://amitu-ted.visualstudio.com/DefaultCollection/iBera/_workitems/edit/11
-async function getUserId(userToken) {
-  return 'demo-user-001';
+async function verifyJwt(jwt){
+  try{
+    var publicKey = await getPuclicKeyFromExchange(jwt);
+    return jsrsasign.jws.JWS.verifyJWT(jwt,publicKey,{alg: ['RS256']});
+  }catch(err){
+    var errMsg = `Error validating Json Web Token: ${util.inspect(err)}`;
+    console.log(err);
+    throw (err);
```

Comment:
you are throwing a string here.
please change to throw new Error(errMsg);

Reply:
fixed